gui/src/components/widgets/cortx-reset-password.vue

```diff
@@ -0,0 +1,203 @@
+          class="cortx-form-group"
+          :class="{
+            'cortx-form-group--error':
+              $v.resetAccountForm.confirmPassword.$error
+          }"
+        >
+          <label
+            class="cortx-form-group-label"
+            for="confirm-password"
+            id="confirm-password-label"
+            >{{ $t("common.confirm-password-label") }}</label
+          >
+          <input
+            class="cortx-form__input_text"
+            type="password"
+            id="confirm-password"
+            name="confirm-password"
+            v-model.trim="resetAccountForm.confirmPassword"
+            @input="$v.resetAccountForm.confirmPassword.$touch"
+          />
+          <span
+            id="confirm-password-notmatch-error"
+            class="cortx-form-group-label cortx-form-group-error-msg"
+            v-if="
+              $v.resetAccountForm.confirmPassword.$dirty &&
+              !$v.resetAccountForm.confirmPassword.sameAsPassword
+            "
+            >{{ $t("common.password-not-match") }}</span
+          >
+        </div>
+      </v-col>
+      <v-col class="col-6 ml-7 pb-6 pt-0">
+        <button
+          type="button"
+          id="reset-password-button"
+          class="cortx-btn-primary"
+          @click="resetPassword()"
+          :disabled="$v.resetAccountForm.$invalid"
+        >
+          {{ $t("login.reset-btn") }}
+        </button>
+      </v-col>
+    </v-card>
+  </v-dialog>
+</template>
+<script lang="ts">
+
+import { Component, Vue, Prop, Watch } from "vue-property-decorator";
+import { Validations } from "vuelidate-property-decorators";
+import { required, sameAs } from "vuelidate/lib/validators";
+import {
+  passwordRegex,
+  passwordTooltipMessage
+} from "./../../common/regex-helpers";
+import { Api } from "../../services/api";
+import apiRegister from "../../services/api-register";
+import i18n from "../preboarding/preboarding.json";
+
+@Component({
+  name: "cortx-reset-password",
+  i18n: {
+    messages: i18n
+  }
+})
+export default class CortxResetPasswrod extends Vue {    
+  public resetAccountForm = {
+    password: "",
+    confirmPassword: ""
+  };  
+  private passwordTooltipMessage = passwordTooltipMessage;
+  
+  @Prop({ default: "" })
+  public authToken: string;
+
+  @Prop({ default: "" })
+  public username: string;
+
+  @Prop({ default: false})
+  public showResetPasswordDialog: boolean
+
+  @Validations()
+  public validations = {
+    resetAccountForm: {
+      password: { required, passwordRegex },
+      confirmPassword: {
+        sameAsPassword: sameAs("password")
+      }
+    }
+  };
+
+  private async resetPassword() {
+    const updateDetails = {
+      confirmPassword: this.resetAccountForm.confirmPassword,
+      password: this.resetAccountForm.password,
+      reset_password: true
+    };
+    this.$store.dispatch(
+      "systemConfig/showLoader",
+      this.$t("login.loading-reset")
+    );
+    const res: any = await Api.patch(
```

Comment:
Please use try/catch to handle error. Pass on success or error status when emitting the event.